napari/layers/utils/_link_layers.py

```diff
@@ -116,11 +126,13 @@ def link_layers(
         if key in _UNLINKERS:
             continue
 
-        def _make_l2_setter(l1=lay1, l2=lay2, attr=attribute):
+        def _make_l2_setter(
+            l1: Layer = lay1, l2: Layer = lay2, attr: str = attribute
+        ) -> Callable:
             # get a suitable equality operator for this attribute type
             eq_op = pick_equality_operator(getattr(l1, attr))
 
-            def setter(event=None):
+            def setter(event: Optional[Event] = None) -> None:
                 new_val = getattr(l1, attr)
                 # this line is the important part for avoiding recursion
                 if not eq_op(getattr(l2, attr), new_val):
```

Comment:
In general, this needs to be refactored. It is not against typing, it is highlighting problematic code part.